.github/workflows/my-ci-pipeline.yml

```diff
@@ -23,28 +23,21 @@ jobs:
     name: Build and image
     runs-on: ubuntu-latest
     needs: build
-    if: github.ref_name == 'module8_updated'
+    if: github.ref_name == 'module-9'
     steps:
     - run: echo "Publishing!"
     - uses: actions/checkout@v2
     - name: docker build
-      run: docker build -t zorion75/todo-app-module8 .
+      run: docker build -t zorion75/todo-app .
     - name: login
       run: echo ${{secrets.DOCKER_HUB_TOKEN}} | docker login --username zorion75 --password-stdin
     - name: push image to docker hub
-      run: docker push zorion75/todo-app-module8
+      run: docker push zorion75/todo-app
 
-  heroku:
-    name: deplay to heroku
+  webhook:
+    name: Azure
     runs-on: ubuntu-latest
     needs: dockerhub
     steps:
-    - uses: actions/checkout@v2
-    - uses: akhileshns/heroku-deploy@v3.12.12
-      with:
-        heroku_api_key: ${{secrets.HEROKU_API}}
-        heroku_app_name: "todo-app-kat-module8"
-        heroku_email: "katarzyna.polonkiewicz@kpmg.co.uk"
-        usedocker: true
-
-          #hello
\ No newline at end of file
+      - name: azure
+        run: curl -dH -X POST ${{secrets.WEBHOOK_URL}}
```

Comment:
I don't think this is quite working as is; the output says:
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100     1    0     0  100     1      0      1  0:00:01 --:--:--  0:00:01     1

But the speed of that is suspicious and I'd expect to see a line like the below:
> {"OperationId":"<guid>","TrackingUrl":"https://<APP_NAME>.scm.azurewebsites.net/api/logstream?filter=op:<guid>,volatile:false"}

That suggests that the WEBHOOK_URL secret might not be set, or might be set wrong. Did you include the escape character (`\`) for the `$` in the url?

Escaping characters can be fiddly, so one way you can potentially work round that if you find it doesn't work is to switch to using an explicit environment variable block, and read the secret in there; I tend to find that fairly reliable. Something like:
```
    needs: dockerhub
    env:
        WEBHOOK_URL: '${{ secrets.WEBHOOK_URL }}'
    steps:
      - name: azure
        run: curl -dH -X POST $WEBHOOK_URL
```
(I believe that if you use the above you won't need/want the escape character, but if that doesn't work then it's worth trying it with!)